File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    private int solve(int n, int[] mem)
    {
        if(n<=2)
        {
            return n;
        }
        if(mem[n]!=-1)
        {
            return mem[n];
        }
        return mem[n]=solve(n-1, mem)+solve(n-2, mem);
    }
    public int climbStairs(int n) {
        int[] mem=new int[n+1];
        Arrays.fill(mem, -1);
        return solve(n, mem);
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def solve(self, n: int, mem: List[int]) -> int:
        if(n<=2):
            return n
        if(mem[n]!= -1):
            return mem[n]
        mem[n]=self.solve(n-1,mem)+self.solve(n-2,mem)
        return mem[n]
        
    def climbStairs(self, n: int) -> int:
        mem=[-1]*(n+1)
        return self.solve(n, mem)